src/extension.ts

```diff
@@ -71,7 +71,19 @@ function npmCheck() {
 
 function handleChange() {
   const installStep = repoType === REPO_TYPE.YARN ? 'yarn' : 'npm install';
-  vscode.window.showInformationMessage(`Lock file changed. Run "${installStep}".`);
+  vscode.window
+    .showInformationMessage(`Lock file changed. Run "${installStep}".`, 'Install dependencies')
+    .then(selection => {
+      if (selection === 'Install dependencies') {
+        vscode.tasks.fetchTasks({ type: 'npm' }).then(tasks => {
```

Comment:
Can you also handle the `yarn install` step

Reply:
In vscode tasks npm is also yarn, I don't know how to explain it but I think vscode handle yarn and npm with the same task. Look : https://share.getcloudapp.com/P8uy0pJ9 my project is under yarn

Reply:
That's why I used task and not the launch of an integrated terminal (that launch either `npm install` or `yarn install`) too

Reply:
Yeah got it. But user previously had to configure the Package manager beforehand for this to work.
```
"npm.packageManager": "yarn"
```
https://github.com/Microsoft/vscode/issues/61203
I am not sure how to handle this from extension. Any suggestions?

We can ask user to add a the above config in his workspace level settings.json.

Reply:
Hmm! I didn't know the task take this settings in consideration! But I tested with both `"npm.packageManager": "yarn"` and `"npm.packageManager": "npm"` and the install was successful too. Maybe it's ok?

Reply:
Yes. Looks good.
This requires user to configure the setting value. Can't we pass a `yarn`/`npm` from extension depending on `yarn.lock`/`package-lock.json` ?
This way the user need not have to set `npm.packageManager` value.

Reply:
Ok I got it. When user does not have this setting and launch the task by clicking the button, if the project is under yarn, it runs `npm install`. I will try to do what you say by writing this configuration if it don't exist in the global config of vscode 👍

Reply:
Can we change global config from extension ?

Can't we configure the task when executing `vscode.tasks.executeTask(task);` by passing either `yarn`/`npm` to `npm ` task. This way extension works irrespective of whether global setting is set or not.

I didn't know about these `tasks` API in `vscode`. So I thought this can be achieved by creating a terminal and executing the corresponding command. If we can achieve this with tasks, that would be great.

Reply:
Yes, we can. I updated the code to check if user does not have the `npm.packageManager` config in global, we add it depends on project repo type. In he had already the `npm.packageManager` global config, it does not override the config so it couldn't be the good config for his current project.

I think task is the good way to achieve this because we can launch `yarn install` or `npm install` juste by setting the right config in his global configuration.